### ISSUE TITLE
Resizing problem in DAIN?

### ISSUE THREAD

**Issue by meguerreroa** [CyFeng16/MVIMP#5]
Hi, i was using your MVIMP DAIN notebook in Google Colab to interpolate a short gif image:
[Original](https://www.pixiv.net/en/artworks/63077093)
I used a 0.25 time_step (original have 8.33 fps so the interpolated should have  33.32) and i got this: 
[Result in MVIMP Colab](https://gfycat.com/plumplimparabianhorse)
It looks like a strobing of different sizes frames.

I tried near the same (to 30 fps) in this [notebook](https://colab.research.google.com/drive/1gzsfDV_MIdehr7Y8ZzWjTuW-mMZRP4Vy) and i got [this](https://gfycat.com/gleefulmintyantlion), less "strobing" (but still there). Looking up into the notebook i found a "RESIZE HOTFIX" method where the interpolated frames are upscaled slightly (apparently, in the interpolation process, they are generated a bit smaller than originals), also found that there are two interpolation methods, INTER_CUBIC and INTER_LANCZOS4

My question is: Could be there a "universal method" to solve this problem? ? Like an algorithm. Is it because of those two interpolation methods? Or maybe is something i'm doing wrong (codecs?) or a problem with DAIN itself.

Thanks in advance.

**Comment by CyFeng16** [comment]
I'll check if it's our problem or not. 
Keep in touch. : )

**Comment by Zotikus1001** [comment]
This is a problem with DAIN it self. 
Some people try to solve it by resizing the interpolated frames.
The "Colab_DAIN.ipynb" in the DAIN repository comes with said hotfix, it does minimize that problem, but it still isnt perfect.

The DAINApp application fixes this completely, but I don't know how.

**Comment by CyFeng16** [comment]
@BrokenSilence Could you please help us improve it according to the "Colab_DAIN.ipynb" in the DAIN repository, I will assist if needed. :smiley_cat:

**Comment by Zotikus1001** [comment]
@CyFeng16 I addded it to inference_dain: https://github.com/BrokenSilence/MVIMP/blob/master/inference_dain.py

But as I said, its not perfect. And I still dont know how to improve it.

**Comment by CyFeng16** [comment]
Many thanks.

**Comment by Zotikus1001** [comment]
The dev from DainApp said this to me: 

> I think it was a pytorch update that change some parameter for rescaling, not sure. Perhaps try to downgrade the pytorch

Maybe give that a try first? 
I will test it when I can, maybe saturday.

**Comment by CyFeng16** [comment]
Some of the advanced features had added into our repo new according to #10 . PTAL.
Looking forward to @BrokenSilence 's further movement.

**Comment by Zotikus1001** [comment]
@CyFeng16 I managed to completely fix this issue.
I did a few things, maybe not all are needed, but it was the only way I could get it to interpolate without errors.

Here is the colab notebook, please take a look: https://github.com/BrokenSilence/MVIMP/blob/master/MVIMP_DAIN_Colab.ipynb

- Downgraded gcc
- Downgraded Torch
- Downgraded Cuda
- Adjusted nvcc_args
- Updated OpenCv

**Comment by CyFeng16** [comment]
@BrokenSilence 
Hi, good day!
Do these changes continue after your partially modified DAIN? (Aka #10 )
I'll try and make a PR as long as everything goes well.
Thank you for your continued support, it means a lot.

**Comment by Zotikus1001** [comment]
@CyFeng16 
I didn't quite understand this part: "Do these changes continue after your partially modified DAIN? (Aka #10 )"

Could you rephrase it?
Thank you.

**Comment by CyFeng16** [comment]
> @CyFeng16 I addded it to inference_dain: https://github.com/BrokenSilence/MVIMP/blob/master/inference_dain.py
> 
> But as I said, its not perfect. And I still dont know how to improve it.

No problem! 
You have already made some modification and I merge it into this PR #10 . Therefore I wonder what you tried at <https://github.com/BrokenSilence/MVIMP/blob/master/MVIMP_DAIN_Colab.ipynb> has connections with the modification you mentioned before or not.
Sorry for the unclear explanation - -||

**Comment by Zotikus1001** [comment]
@CyFeng16 
Thank you.
It's a completely different solution. So I did remove the frame resizing hotfix from inference_dain: https://github.com/BrokenSilence/MVIMP/commit/a5297bc74ad4ebf230873e9b9fa3cf728f46c82e

That was all. :)

**Comment by CyFeng16** [comment]
@BrokenSilence Thanks in advance. :+1:
